### ISSUE TITLE
max-line-length should not count each tab as one character when indent is set to tabs

### ISSUE THREAD

**Issue by asfernandes** [palantir/tslint#4090]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.9.1
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
			menu.subMenus
				.filter(subMenu => subMenu)
				.forEach(subMenu => scan(sistema, menu.label ? `${prefix}${menu.descSistema ? menu.descSistema : menu.label} > ` : '', subMenu));
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:recommended",
  "rules": {
    "max-line-length": {
      "options": [140],
    },
    "indent": [true, "tabs", 4],
  }
}
```

#### Actual behavior

It does not count the line as more than 140 characters.

#### Expected behavior

Last line should show as an error violating max-line-length rule.

**Comment by asfernandes** [comment]
Sorry, this was a problem when I editted my json to just put the relevant part. Here is my full file:

```json
{
  "extends": "tslint:recommended",
  "rules": {
    "max-line-length": {
      "options": [140]
    },
    "indent": [true, "tabs", 4],
    "no-trailing-whitespace": [false],
    "one-line": [false],
    "arrow-parens": [true, "ban-single-arg-parens"],
    "trailing-comma": [true,
      {
        "multiline": "never",
        "singleline": "never"
      }
    ],
    "object-literal-sort-keys": [false],
    "no-consecutive-blank-lines": [true, 2],
    "interface-name": [false],
    "comment-format": [false],
    "member-access": false,
    "array-type": false,
    "max-classes-per-file": false,
    "align": [true, "elements", "members", "statements"],
    "member-ordering": [false],
    "no-namespace": false,
    "no-var-requires": false,
    "no-implicit-dependencies": true,

    "curly": [false],
    "no-conditional-assignment": false,
    "ordered-imports": [false],
    "quotemark": [true, "single"],
    "triple-equals": [false]
  }
}
```

**Comment by asfernandes** [comment]
VSCode 1.25.1 with eg2.tslint

**Comment by johnwiseheart** [comment]
I just looked at this again:

```typescript
			menu.subMenus
				.filter(subMenu => subMenu)
				.forEach(subMenu => scan(sistema, menu.label ? `${prefix}${menu.descSistema ? menu.descSistema : menu.label} > ` : '', subMenu));
```

The last line in your example is only `133` characters, since you're using tabs.

**Comment by asfernandes** [comment]
A tab should count as 4 characters with `"indent": [true, "tabs", 4]` as VSCode does too. That's the issue.

**Comment by asfernandes** [comment]
Of course it's not just "count as 4 characters". it should match the indentation we see in the screen.

**Comment by johnwiseheart** [comment]
Ooh I see - this is kinda unclear. Currently `"indent": [true, "tabs", 4]` does not mean convert tabs to spaces before running other rules. The docs are a little unclear:

```
A second optional argument specifies indentation size:
* 2 enforces 2 space indentation.
* 4 enforces 4 space indentation.

Indentation size is required for auto-fixing, but not for rule checking.
```

I believe that right now, the `indent` rule is only checking that you use either tabs, or _size_ number of spaces.

**Comment by johnwiseheart** [comment]
It's probably worth improving upon the documentation for the `indent` rule to specify this.

**Comment by asfernandes** [comment]
I understand that each rule has a objective, but max-line-length counting tabs as 1 does not make any sense.

**Comment by johnwiseheart** [comment]
What makes you say that? A tab _is_ one character, `\t`. Rules run independently, so the `max-line-length` rule would have to have some option to treat tabs as multiple spaces for this to work how you're expecting. A tab on one system can be a different size to another system, so its hard to just assign one value to us. I'm not necessarily against this idea, but as it is right now I think the rule makes sense.

**Comment by JoshuaKGoldberg** [comment]
@asfernandes that would be great actually :)

**Comment by rwaskiewicz** [comment]
@johnwiseheart 

> I believe that right now, the indent rule is only checking that you use either tabs, or size number of spaces.

I started to take a look at this this evening,  the following line in `IndentRule.ts` confuses me:
https://github.com/palantir/tslint/blob/237fd905a9fb3c9ca9325d2da6e8145e4a44fe51/src/rules/indentRule.ts#L105-L106

It looks like if we're using tabs, we create a regex with N number of space characters (where N is the value provided in the second param) but the error message consists of tabs?  I would have expected the true/false components of that first ternary to be switched. Maybe I've confused myself on the documentation and implementation, is there something I'm missing?

**Comment by JoshuaKGoldberg** [comment]
One of the goals of `max-line-length` is to strop things lines from being too long horizontally & wrapping or overflowing past thin displays. It seems like treating tabs as 1-character sized goes agains that. Thoughts @johnwiseheart? Feels like taking in a bugfix to allow tabs to count as the larger size in rule checking, while technically a breaking change, might be an improvement for the rule?

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
